src/shared/cloudformation/templateRegistry.ts

```diff
@@ -125,18 +128,101 @@ export class CloudFormationTemplateRegistry {
 }
 
 /**
- * Helper function that returns an map of resource names to CloudFormation.Resource objects.
- * Unlike a CloudFormation.TemplateResources object, all resources in this array are guaranteed to be defined.
- * @param templateDatum TemplateDatum object to extract resources from
+ * Filters an array of TemplateDatum objects to those that are tied to the given filepath and handler function.
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param unfilteredTemplates Array containing TemplateDatum objects to filter
  */
-export function getResourcesFromTemplateDatum(templateDatum: TemplateDatum): Map<string, CloudFormation.Resource> {
-    const map = new Map<string, CloudFormation.Resource>()
-    for (const resourceKey of Object.keys(templateDatum.template.Resources!)) {
-        const resource = templateDatum.template.Resources![resourceKey]
-        if (resource) {
-            map.set(resourceKey, resource)
+export function getTemplatesAssociatedWithHandler(
+    filepath: string,
+    handler: string,
+    unfilteredTemplates: TemplateDatum[] = CloudFormationTemplateRegistry.getRegistry().registeredTemplates
+): TemplateDatum[] {
+    // find potential matching templates
+    return unfilteredTemplates.filter(templateDatum => {
+        return !!getResourceAssociatedWithHandlerFromTemplateDatum(filepath, handler, templateDatum)
+    })
+}
+
+/**
+ * Returns the first Cloudformation Resource in a TemplateDatum that is tied to the filepath and handler given.
+ * Returns undefined if none are found.
+ * TODO: Is there a reason to return more than the first resource?
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param templateDatum TemplateDatum object to search through
+ */
+export function getResourceAssociatedWithHandlerFromTemplateDatum(
+    filepath: string,
+    handler: string,
+    templateDatum: TemplateDatum
+): CloudFormation.Resource | undefined {
+    const templateDirname = path_.dirname(templateDatum.path)
+    // template isn't a parent or sibling of file
```

Comment:
If template files reside elsewhere in the workspace but (CodeUri) reference a location that includes this filepath, are they also suitable candidates?

Reply:
Can template files use an absolute path? That's the only reason I could see that this would work; I've only seen templates using relative paths.

Reply:
Yes